src/plot_api/subroutines.js

```diff
@@ -613,6 +612,61 @@ exports.drawData = function(gd) {
     return Plots.previousPromises(gd);
 };
 
+// Draw (or redraw) all traces in one go,
+// useful during drag and selection where buffers of targeted traces are updated,
+// but all traces need to be redrawn following clearGlCanvases.
+//
+// Note that _module.plot for regl trace does NOT draw things
+// on the canvas, they only update the buffers.
+// Drawing is perform here.
+//
+// TODO try adding per-subplot option using gl.SCISSOR_TEST for
+// non-overlaying, disjoint subplots.
+//
+// TODO try to include parcoords in here.
```

Comment:
Does it break if a parcoords trace is mixed in with other regl traces? Or is this TODO just an optimization?

Reply:
> Does it break if a parcoords trace is mixed in with other regl traces

It's no worse than it is now. parcoords + other regl trace works fine on first-render, but breaks during cartesian drag and selection.

Reply:
This was discussed during the regl-based scattergl push, but I couldn't find a dedicated issue on this topic, so -> https://github.com/plotly/plotly.js/issues/3069